Fabric: add minimal ProgressView component for iOS

ericlewis opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. The three fields below are mandatory. -->

## Summary

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
Codegen'd ProgressView, this brings RNTester’s example back to life!

<img src="https://i.imgur.com/bPAESJT.png" width="300" />

## Changelog

<!-- Help reviewers and the release process by writing your own changelog entry. See http://facebook.github.io/react-native/docs/contributing#changelog for an example. -->

[iOS] [Added] - Fabric: ProgressView

## Test Plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->
View ProgressViewIOS example in RNTester with Fabric enabled.

shergin commented on React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.h:
```diff
@@ -0,0 +1,21 @@
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * UIView class for root <ProgressView> component.
```
--root--

ericlewis replied:
@shergin all the components are like this lol
```
/**
 * UIView class for root <Slider> component.
 */
```

shergin commented on React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.mm:
```diff
@@ -0,0 +1,64 @@
+
+  // `trackTintColor`
+  if (oldProgressViewProps.trackTintColor != newProgressViewProps.trackTintColor) {
+    _progressView.trackTintColor = [UIColor colorWithCGColor:newProgressViewProps.trackTintColor.get()];
```
`.get` is an implementation detail, we have to use special conversion function from RCTConvertions.h`.

ericlewis commented:
@shergin I think that addresses your feedback, unless you want me to remove root- but other components have that too, so maybe I could do that in a diff PR..

fkgozali commented:
In the spirit of Lean Core, I'm not sure we want this in core? So I think we can hold off for now?

cpojer commented:
Closing based on @fkgozali's comment. Thanks for experimenting with all the new awesome features :)
